src/main/kotlin/com/cognifide/gradle/aem/deploy/tasks/AbstractTask.kt

```diff
@@ -169,4 +176,79 @@ abstract class AbstractTask : DefaultTask(), AemTask {
         return response
     }
 
+    protected fun deletePackage(installedPackagePath: String, sync: DeploySynchronizer) {
+        val url = sync.htmlTargetUrl + installedPackagePath + "/?cmd=delete"
+
+        logger.info("Deleting package using command: " + url)
+
+        try {
+            val rawHtml = sync.post(url)
+            val response = DeleteResponse(rawHtml)
+
+            when (response.status) {
+                AbstractHtmlResponse.Status.SUCCESS,
+                AbstractHtmlResponse.Status.SUCCESS_WITH_ERRORS -> if (response.errors.isEmpty()) {
+                    logger.info("Package successfully deleted.")
+                } else {
+                    logger.warn("Package deleted with errors.")
+                    response.errors.forEach { logger.error(it) }
+                    throw DeployException("Package deleted with errors!")
+                }
+                AbstractHtmlResponse.Status.FAIL -> {
+                    logger.error("Package deleting failed.")
+                    response.errors.forEach { logger.error(it) }
+                    throw DeployException("Package deleting failed!")
+                }
+            }
+
+        } catch (e: Exception) {
+            throw DeployException("Cannot delete package.", e)
+        }
+    }
+
+    protected fun uninstallPackage(installedPackagePath: String, sync: DeploySynchronizer) {
+        val url = sync.htmlTargetUrl + installedPackagePath + "/?cmd=uninstall"
+
+        logger.info("Uninstalling package using command: " + url)
+
+        try {
+            val rawHtml = sync.post(url, mapOf(
+                    "recursive" to config.recursiveInstall,
+                    "acHandling" to config.acHandling
+            ))
+            val response = UninstallResponse(rawHtml)
+
+            when (response.status) {
+                AbstractHtmlResponse.Status.SUCCESS,
```

Comment:
agreed to leave as it is, with AbstractHtmlResponse name changed to HtmlResponse